src/selectors/entities/teams.js

```diff
@@ -115,7 +115,7 @@ export const getMyTeams = createSelector(
     getTeams,
     getTeamMemberships,
     (teams, members) => {
-        return Object.values(teams).filter((t) => members[t.id]);
+        return Object.values(teams).filter((t) => members[t.id] && t.delete_at === 0);
```

Comment:
this selector help with the problem when the user gets the websocket event but if the event is missed for some reason `delete_at` will probably not be set.

in my opinion this we should also remove the memberships in the reducer when getting the response of `getMyTeamMembers`

Reply:
> in my opinion this we should also remove the memberships in the reducer when getting the response of getMyTeamMembers

The payload for `RECEIVED_MY_TEAM_MEMBERS` does not contain the actual teams, so the reducer cannot determine which teams have been archived when it receives the action. This is actually the main issue. The reducer has no choice but to trust that all of the memberships are valid.

https://github.com/mattermost/mattermost-redux/blob/871d2d248a618c100d083dd95c3151de8be6f539/src/reducers/entities/teams.js#L85

If you wanted to filter them for that action, it would have to be pre-dispatch in the action function, which doesn't seem right.

Reply:
why it does not seem right? maybe the action data needs to have the members and the teams or whatever it is to determine that the user is no longer a member of.

by not applying the right memberships and removing the deleted ones it can result in unexpected behaviors as it is now.